### PR TITLE
Add ability to filter out system attributes

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

File: container/container.go [hunk 1]
```diff
@@ -329,14 +330,26 @@ func (x Container) Attribute(key string) string {
 // IterateAttributes iterates over all Container attributes and passes them
 // into f. The handler MUST NOT be nil.
 //
-// See also SetAttribute, Attribute.
+// See also SetAttribute, Attribute, IterateNonSystemAttributes.
```
Comment (roman-khimov) [inline container/container.go:333]: These should be doc links.

File: container/container.go [hunk 2]
```diff
@@ -329,14 +330,26 @@ func (x Container) Attribute(key string) string {
...
 	attrs := x.v2.GetAttributes()
 	for i := range attrs {
 		f(attrs[i].GetKey(), attrs[i].GetValue())
 	}
 }
 
+// IterateNonSystemAttributes iterates non-system Container attributes and
+// passes them into f. The handler MUST NOT be nil.
+//
+// See also SetAttribute, Attribute, IterateAttributes.
```
Comment (roman-khimov) [inline container/container.go:344]: These as well.

File: object/object.go [hunk 3]
```diff
@@ -335,7 +336,9 @@ func (o *Object) SetPayloadHomomorphicHash(v checksum.Checksum) {
 	})
 }
 
-// Attributes returns object attributes.
+// Attributes returns all object attributes.
+//
+// See also SetAttributes, NonSystemAttributes.
```
Comment (roman-khimov) [inline object/object.go:341]: And here.

File: object/object.go [hunk 4]
```diff
@@ -350,6 +353,25 @@ func (o *Object) Attributes() []Attribute {
 	return res
 }
 
+// NonSystemAttributes returns non-system object attributes.
+//
+// See also SetAttributes, Attributes.
```
Comment (roman-khimov) [inline object/object.go:358]: And here.

File: object/object.go [hunk 5]
```diff
@@ -350,6 +353,25 @@ func (o *Object) Attributes() []Attribute {
...
+func (o *Object) NonSystemAttributes() []Attribute {
```
Comment (roman-khimov) [inline object/object.go:359]: We have a notion of "user attributes" in the spec, maybe it'd be better than "non-system attributes"?

Comment (cthulhu-rider) [reply object/object.go:359]: thought about that. [This](https://github.com/nspcc-dev/neofs-api/blob/7ae1a7438b19fc4ca32c451db67de170edb539a7/object/types.proto#L156-L157) bothers me a bit. Mb u mentioned some other spec piece?

Comment (roman-khimov) [reply object/object.go:359]: More like these
https://github.com/nspcc-dev/neofs-spec/blob/master/01-arch/01-netmap.md#network-map
https://github.com/nspcc-dev/neofs-spec/blob/master/01-arch/07-acl.md#extended-acl

Comment (cthulhu-rider) [reply object/object.go]: name `User`

File: container/container.go [hunk 6]
```diff
@@ -329,14 +330,26 @@ func (x Container) Attribute(key string) string {
...
+// See also [Container.SetAttribute], [Container.Attribute], [Container.IterateUserAttributes].
 func (x Container) IterateAttributes(f func(key, val string)) {
 	attrs := x.v2.GetAttributes()
 	for i := range attrs {
 		f(attrs[i].GetKey(), attrs[i].GetValue())
 	}
 }
 
+// IterateUserAttributes iterates over user attributes of the Container and
+// passes them into f. The handler MUST NOT be nil.
```
Comment (smallhive) [inline container/container.go:342]: I consider this _The handler MUST NOT be nil._ is obvious

**Review by smallhive — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**